parsing/location.mli

```diff
@@ -99,7 +99,39 @@ val rewrite_absolute_path: string -> string
         variable (https://reproducible-builds.org/specs/build-path-prefix-map/)
         if it is set. *)
 
+val rewrite_exists: string -> string option option
+(** [rewrite_exists path] uses a BUILD_PATH_PREFIX_MAP mapping
+    (https://reproducible-builds.org/specs/build-path-prefix-map/)
+    and tries to find a source in mapping
+    that maps to a result that exists in the file system.
+    There are the following return values:
+    - None, means BUILD_PATH_PREFIX_MAP is not set.
+    - Some None, no source prefixes of [path] in the mapping were found,
+    - Some (Some target), means target is the first file (in priority
+      order) that [path] mapped to that exists in the file system.
+    - Not_found raised, means some source prefixes in the map
+      were found that matched [path], but none of them existed
+      in the file system. *)
+
+val matching_dirs: string -> string list option
+(** [matching_dirs absdir] accumulates a list of existing directories,
+    [dirs], that are the result of mapping abstract directory, [absdir],
+    over all the mapping pairs in the BUILD_PATH_PREFIX_MAP environment
+    variable. The list [dirs] will be in priority order (head as highest
+    priority). The possible results are:
+    - None, means BUILD_PATH_PREFIX_MAP is not set.
+    - Some [], no source prefixes in the mapping were found,
+    - Some dirs, means dirs are the directories found
+    - Not_found raised, means some source prefixes in the map
+      were found that matched [path], but none of mapping results
+      were existing directories.
+
+    See the BUILD_PATH_PREFIX_MAP spec at
+    (https://reproducible-builds.org/specs/build-path-prefix-map/)
+    *)
+
 val absolute_path: string -> string
+    (** Makes absolute and uses rewrite_absolute_path. *)
```

Comment:
We now have many BUILD_PATH_PREFIX_MAP functions in this module. I think they would deserve their own documentation section (currently they are in `{1 Printing locations}`, they could move to `{1 Rewriting paths}` or something). We could also think about their naming as a group, instead of individual functions.

- should all functions that apply some rewriting start with the `rewrite_` prefix? Then `absolute_path` should change now. (But maybe we don't want that.)
- `rewrite_exists` and `matching_dirs` are closely related but have inconsistent names; maybe `rewrite_find_first_existing` and `rewrite_find_all_existing_dirs`?

Reply:
Done